### PR TITLE
Add vendored `linux/arm64` FIPS binaries

### PR DESCRIPTION
**Pull request by bleggett**
I don't think vendoring our own FIPS binaries is a great long term solution, but this should at least achieve parity with the current `x86` approach and let us build FIPS `ztunnel` binaries for the two most common architectures contributors are likely to have and that we will ultimately need to generate containerized artifacts for. Also updated README accordingly.

- I still think we probably should not be defining `BORING_BSSL` env vars in `.cargo/config.toml` - instead we should expect people to pass them as prefix args to `cargo build/run`.
- If we want to avoid that, we will need to add dynamic platform-based vendored path selection logic to `build.rs` in https://github.com/cloudflare/boring  - because of how their `build.rs` currently works, we cannot do that dynamic selection in our own `build.rs` - I've tried. Arguably that is something we should fix in upstream.
- We still actually need to do arm64 FIPS CI builds (https://github.com/istio/ztunnel/issues/357), and until the above are resolved we cannot.

Some of the hassle around letting https://github.com/cloudflare/boring auto-build `boringssl` itself should get a bit easier once the upstream PRs from @nmittler and @PiotrSikora are merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,15 +33,45 @@ to ensure that compliance is granted only to the exact binary tested.
...
+We vendor FIPS boringssl binaries for 
+- `linux/x86_64`
+- `linux/arm64`
+
+To use these vendored libraries and build ztunnel for either of these OS/arch combos, for the moment you must manually edit
+[.cargo/config.toml](.cargo/config.toml) and change the values of BORING_BSSL_PATH and BORING_BSSL_INCLUDE_PATH under the `[env]` key to match the path to the vendored libraries for your platform, e.g:
+
+#### For linux/x86_64
+``` toml
+BORING_BSSL_PATH = { value = "vendor/boringssl-fips/linux_x86_64", force = true, relative = true }
```
Comment (howardjohn) [inline README.md:48]: Do you think its feasible to get the boring crate to allow some form of substition so we can just put `vendor/boringssl-fips/{platform}` or whatever?

File: README.md [hunk 2]
```diff
@@ -30,15 +32,45 @@ to ensure that compliance is granted only to the exact binary tested.
...
+BORING_BSSL_INCLUDE_PATH = { value = "vendor/boringssl-fips/linux_x86_64/include/", force = true, relative = true }
+```
+
+#### For linux/arm64
+
+``` toml
+BORING_BSSL_PATH = { value = "vendor/boringssl-fips/linux_arm64", force = true, relative = true }
+BORING_BSSL_INCLUDE_PATH = { value = "vendor/boringssl-fips/linux_arm64/include/", force = true, relative = true }
```
Comment (howardjohn) [inline README.md:56]: nit: is the include/ arch specific or can we share it?

**Review by howardjohn — COMMENTED** [review]
LGTM! does cross compile work?

**Comment by bleggett** [comment]
> LGTM! does cross compile work?

Not tried it - just verified that a FIPS arm64 build works within `make shell`. If that works, CC ought to.

File: README.md (hunk 1 above)
Comment (bleggett) [reply README.md:48]: Possibly/maybe - I think once the PRs @PiotrSikora and @nmittler already have outstanding in that repo get merged, that's something I'll look at if no one beats me to it.

File: README.md (hunk 2 above)
Comment (bleggett) [reply README.md:56]: Yeah I thought about that, I think so, lemme give that a go.

Comment (howardjohn) [reply README.md:56]: I put a hold in case you want to do this but approved, feel free to remove

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
cross compile failing in linking for me, but I may have done something wrong

**Comment by howardjohn** [comment]
actually I guess you need to manually st the linker: CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc cargo build --target=aarch64-unknown-linux-gnu

File: README.md (hunk 2 above)
Comment (bleggett) [reply README.md]: Yep - consolidated, ty for asking.

**Comment by bleggett** [comment]
> actually I guess you need to manually st the linker: CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc cargo build --target=aarch64-unknown-linux-gnu

Generally speaking I find multiarch `buildx`  targets way easier to get working than CC - good to know.

**Comment by howardjohn** [comment]
> > actually I guess you need to manually st the linker: CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc cargo build --target=aarch64-unknown-linux-gnu
> 
> Generally speaking I find multiarch `buildx` targets way easier to get working reliably/reproducibly than CC - good to know.

Good point, wasn't sure how slow it was.. In envoy we ended up using dedicated arm64 nodes because envoy takes forever to build. Here we have more flexibility, can cross compile, qemu, or use dedicated nodes - whatever is easy/fast

**Pull request merged**